src/lesson24/js/main.js

```diff
@@ -0,0 +1,149 @@
+  target.focus();
+  target.addEventListener("keydown", (event) => {
+    if (event.key === "Tab") {
+      event.preventDefault();
+      const focusableElements = [
+        ...target.querySelectorAll(focusableElementsSelector),
+      ];
+      const currentFocusedItemIndex = focusableElements.indexOf(
+        document.activeElement
+      );
+      const nextFocusedItemIndex =
+        (currentFocusedItemIndex + 1) % focusableElements.length;
+      focusableElements[nextFocusedItemIndex].focus();
+    }
+  });
+}
+
+tabAccessControl(document.getElementById("js-form"));
+document.getElementById("js-userName").focus();
+
+const agreeButton = document.getElementById("js-agreeButton");
+function enableAgreeButton() {
+  agreeButton.className =
+    "bg-blue-300 text-white font-bold rounded px-4 py-2 mb-20 hover:bg-blue-700";
+  agreeButton.disabled = false;
+}
+
+function addClickAgreeToCloseModal() {
+  agreeButton.addEventListener("click", () => {
+    closeModal();
+    enableSubmitButton();
+  });
+}
+
+const submitButton = document.getElementById("js-submitButton");
+function enableSubmitButton() {
+  submitButton.disabled = false;
+  submitButton.className =
+    "w-full bg-blue-300 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded-lg transition-colors";
+}
+
+function disableSubmitButton() {
+  submitButton.disabled = true;
+  submitButton.className =
+    "w-full bg-gray-300 text-white font-bold py-2 px-4 rounded-lg transition-colors";
+}
+
+function checkboxesToBeChecked() {
+  const agreeCheckBoxes = document.querySelectorAll('[data-id="js-checkbox"]');
+  agreeCheckBoxes.forEach((agreeCheckbox) => {
+    agreeCheckbox.disabled = false;
+    agreeCheckbox.checked = true;
+  });
+}
+
+const agreeCheckbox = document.getElementById("js-agreeCheckbox");
+function toggleAgreeCheckbox() {
+  agreeCheckbox.addEventListener("change", () => {
+    agreeButton.disabled = !agreeCheckbox.checked;
+    if (agreeCheckbox.checked) {
+      agreeButton.className =
+        "bg-blue-300 text-white font-bold rounded px-4 py-2 mb-20 hover:bg-blue-700";
+    } else {
+      agreeButton.className =
+        "bg-gray-300 text-white font-bold rounded px-4 py-2 mb-20";
+    }
+  });
+}
+
+const submitCheckbox = document.getElementById("js-submitCheckbox");
+function toggleSubmitCheckbox() {
+  submitCheckbox.addEventListener("change", (e) => {
+    submitButton.disabled = !submitCheckbox.checked;
+    if (submitCheckbox.checked) {
+      enableSubmitButton();
+    } else {
+      disableSubmitButton();
+    }
+  });
+}
+
+function changeAgreeToClickable(entries) {
+  if (!entries[0].isIntersecting) {
+    return;
+  }
+  enableAgreeButton();
+  addClickAgreeToCloseModal();
+  checkboxesToBeChecked();
+  toggleAgreeCheckbox();
+  toggleSubmitCheckbox();
+}
+
+document.getElementById("js-submitButton").addEventListener("click", (e) => {
```

Comment:
IMO
Variables already defined can be used
```suggestion
submitButton.addEventListener("click", (e) => {
```

Reply:
Thank you sae-san :)
I removed the duplicate code.
f4126aa7dd441257102f96f724e07819c513dab4